src/ParametricOptInterface.jl

```diff
@@ -297,17 +345,25 @@ end
 #     MOI.get(model.optimizer, attr)
 # end
 
-# TODO
-# You might have transformed quadratic functions into affine functions so this is incorrect
-# function MOI.get(model::ParametricOptimizer, ::MOI.ListOfConstraints)
-#     constraints = Set{Tuple{DataType, DataType}}()
-#     inner_ctrs = MOI.get(model.optimizer, MOI.ListOfConstraints())
-#     for (F, S) in inner_ctrs
-#         push!(constraints, (F,S))
-#     end
-
-#     collect(constraints)
-# end
+function MOI.get(model::ParametricOptimizer, ::MOI.ListOfConstraints)
+    inner_ctrs = MOI.get(model.optimizer, MOI.ListOfConstraints())
+    !has_quadratic_constraint_caches(model) && return inner_ctrs
+
+    cache_keys = collect(keys(model.quadratic_added_cache))
+    constraints = Set{Tuple{DataType,DataType}}()
+
+    for (F, S) in inner_ctrs
+        inner_index =
+            MOI.get(model.optimizer, MOI.ListOfConstraintIndices{F,S}())
+        cache_map_check =
+            quadratic_constraint_cache_map_check(mode, inner_index)
```

Comment:
In the `AbstractBridgeOptimizer`, we collect all the possible constraint types and them filter with NumberOfConstraints. If NumberOfConstraints is zero then we remove it from the list.
Here, you can look over `keys(quadratic_added_cache)` and add the `F`-`S` types of all the keys in `constraints`.
To implement `NumberOfConstraints`, you call `NumberOfConstraints` to the inner optimizer. Then you remove the number of constraints of that that in `values(quadratic_added_cache)`